src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -84,6 +85,26 @@ internal static Summary[] Run(BenchmarkRunInfo[] benchmarkRunInfos)
 
                     foreach (var benchmarkRunInfo in supportedBenchmarks) // we run them in the old order now using the new build artifacts
                     {
+                        if (benchmarkRunInfo.Config.Options.IsSet(ConfigOptions.Resume))
+                        {
+                            var directoryInfo = new DirectoryInfo(rootArtifactsFolderPath);
+                            var lastFiles = directoryInfo.GetFiles($"{title.Split('-')[0]}*");
+                            if (lastFiles.Count() > 1)
+                            {
+                                var lastUpdatedFile = lastFiles[lastFiles.Length - 2];
+                                var text = File.ReadAllText(lastUpdatedFile.FullName);
+                                var regex = new Regex("--benchmarkId (.*?) in");
+                                var match = regex.Match(text);
+                                if (match.Success)
+                                {
+                                    var parsedId = int.Parse(match.Groups[1].Value);
```

Comment:
the discovery of `parseId` should be moved out of the loop, otherwise we are going to repeat it for all benchmarks and we want to do it only once

Reply:
> the discovery of `parseId` should be moved out of the loop, otherwise we are going to repeat it for all benchmarks and we want to do it only once

In the loop there is `benchmarkRunInfo` and I can know if the option `resume` is set. 
If I follow your suggestion, Do you have any idea how can I do this? (know if set resume out of loop)

Reply:
> Do you have any idea how can I do this? (know if set resume out of loop)

Before the loop starts, you can check if any of the benchmarks present in `supportedBenchmarks` has this flag set to true.

Something like this:


```cs
if (supportedBenchmarks.Any(benchmark => benchmark.Config.Options.IsSet(ConfigOptions.Resume))
```

I know it's not obvious that we don't check it for all of the benchmarks, but this is something that we frequently do in BDN.